packages/airnode-admin/src/implementation.ts

```diff
@@ -35,6 +36,62 @@ export const parseTransactionOverrides = (args: Arguments): ethers.Overrides =>
   return overrides;
 };
 
+/**
+ * Parses Ethers transaction override options and sets them based on the transaction type supported by the network and user inputs.
+ * @param provider Ethers provider
+ * @param overrides Overrides object passed into the function
+ * @returns The parsed overrides object with values compatible with Ethers
+ */
+export const parseOverrides = async (
+  provider: ethers.providers.Provider,
+  overrides: ethers.Overrides = {}
+): Promise<ethers.Overrides> => {
+  const { gasPrice, maxFeePerGas, maxPriorityFeePerGas } = overrides;
+
+  const feeData = await provider.getFeeData();
+  const supportsEip1559 = !!(feeData.maxFeePerGas && feeData.maxPriorityFeePerGas);
+  const hasEip1559Overrides = !!(maxFeePerGas || maxPriorityFeePerGas);
```

Comment:
Note that with this check using `maxFeePerGas || maxPriorityFeePerGas` instead of `maxFeePerGas && maxPriorityFeePerGas` will not cause `parseOverrides` to throw, but ethers will end up throwing because `maxFeePerGas` has not been defined?

I used `||` so we can use the same check in the if-blocks below.

Reply:
If the user is specifying overrides then they should understand the effect of those overrides :shrug: :laughing: